core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -248,6 +249,18 @@ public long getPeakMemoryUsedBytes() {
   }
 
   /**
+   * @return the total amount of time spent sorting data (in-memory only).
+   */
+  public long getSortTimeNanos() {
+    long totalTime = totalSortTimeNanos;
+    UnsafeInMemorySorter sorter = inMemSorter;
+    if (sorter != null) {
```

Comment:
There is very small chance to double-count it when this is called between `totalSortTimeNanos += inMemSorter.getSortTimeNanos()` and `getSortTimeNanos()`

Since we have only one UnsafeInMemorySorter, we should return either `inMemSorter.getSortTimeNanos()` or `totalSortTimeNanos `